### DISCUSSION TITLE
Example for absolute views for black litterman model?

### DISCUSSION THREAD

**Discussion by aliencaocao** [dcajasn/Riskfolio-Lib#124] [Q&A]
Hi, I have seen https://nbviewer.org/github/dcajasn/Riskfolio-Lib/blob/master/examples/Tutorial%203.ipynb but I am still not sure how do I formulate the views df if I want to use absolute views, since the example is relative views. Could you please provide an example of the views df I'm supposed to prepare? Thank you.

**Comment by kfuangsung** [comment]
Referred to
https://riskfolio-lib.readthedocs.io/en/latest/constraints.html?highlight=asset_views#ConstraintsFunctions.assets_views
you suppose to leave 'Type Relative', 'Relative Set' and 'Relative' as blank

**Comment by heykav** [comment]
@kfuangsung is right on the fields to leave blank - here's a concrete, purely-absolute-views example built from that, since the fields alone can be hard to picture without a worked case.

Say you have 3 assets and want two absolute views: META >= 3%, WFC <= 30%.

```python
import pandas as pd
import riskfolio as rp

assets = ["META", "WFC", "TLT"]
asset_classes = pd.DataFrame({"Assets": assets})

views = pd.DataFrame({
    "Disabled": [False, False],
    "Type": ["Assets", "Assets"],
    "Set": ["", ""],
    "Position": ["META", "WFC"],
    "Sign": [">=", "<="],
    "Return": [0.03, 0.30],
    "Type Relative": ["", ""],
    "Relative Set": ["", ""],
    "Relative": ["", ""],
})

P, Q = rp.assets_views(views, asset_classes)
```

Each row is one absolute view because `Type Relative`/`Relative Set`/`Relative` are all empty strings, not because of anything else in the row - that's the one field group that actually switches a row between relative and absolute in `assets_views()` (`riskfolio/src/ConstraintsFunctions.py`). The full docstring example on that function mixes relative and absolute views in one dataframe (its 4th row, META, is an absolute view exactly like this) if you want to see both side by side.
